### PR TITLE
 `GenericGraph`: add `copy()` and `move()` methods to `ForwardNode`, `BidirectionalNode` and `MutableEdgeNode`

### PR DESCRIPTION
**Pull request by Cvelth**

### CODE REVIEW

**Comment by pfez** [comment]
LGTM

File: include/revng/ADT/GenericGraph.h [hunk 1]
```diff
@@ -226,6 +227,9 @@ class ForwardNode
   ForwardNode(const ForwardNode &) = default;
   ForwardNode(ForwardNode &&) = default;
 
+  NodeData copy() const { return *this; };
+  NodeData &&move() { return std::move(*this); };
```
Comment (aleclearmind) [inline include/revng/ADT/GenericGraph.h:231]: These methods might be a bit suspicious in the context of the class that inherits from it.
What about `copyData`/`moveData`?

Comment (Cvelth) [reply include/revng/ADT/GenericGraph.h]: Done.

**Comment by aleclearmind** [comment]
You forgot the blank line in the commit message.
Merged in 6714c4c. Thanks!

**Pull request closed without merging**